test/plugin/test_sd_dns.rb

```diff
@@ -74,9 +74,10 @@ def config(param: nil, af: 'ipv4')
           address_family ipv4
         </entry>
       ]))
-      assert_equal Fluent::Plugin::ServiceDiscovery::Service.new(:dns, '127.0.0.1', 80, '0-localhost', 60, false, '', '', nil), @sd_dns.services[0]
-      assert_equal Fluent::Plugin::ServiceDiscovery::Service.new(:dns, '127.0.0.1', 81, '0-localhost', 60, false, '', '', nil), @sd_dns.services[1]
-      assert_equal Fluent::Plugin::ServiceDiscovery::Service.new(:dns, '127.0.0.1', 82, '0-localhost', 60, false, '', '', nil), @sd_dns.services[2]
+      services = @sd_dns.services.sort_by {|e| e.name }
+      assert_equal Fluent::Plugin::ServiceDiscovery::Service.new(:dns, '127.0.0.1', 80, '0-localhost', 60, false, '', '', nil), services[0]
+      assert_equal Fluent::Plugin::ServiceDiscovery::Service.new(:dns, '127.0.0.1', 81, '0-localhost', 60, false, '', '', nil), services[1]
+      assert_equal Fluent::Plugin::ServiceDiscovery::Service.new(:dns, '127.0.0.1', 82, '0-localhost', 60, false, '', '', nil), services[2]
```

Comment:
BTW although most  tests of fluentd and plugins split assertions for an array to each elements, I think it should be done by one assertion:

```
assert_equal(expected_array, services)
```

because an assertion stops the tests and remaining assertions aren't evaluated.